tests/unit/test_charm.py

```diff
@@ -243,64 +253,72 @@ def test_given_config_file_not_written_when_start_simulation_action_then_event_f
 
         event.fail.assert_called_with(message="Config file is not written")
 
-    @patch("ops.model.Container.exec")
     def test_given_simulation_command_fails_with_execerror_when_start_simulation_action_then_event_fails(  # noqa: E501
-        self, patch_exec
+        self,
     ):
         self.harness.add_storage("config", attach=True)
         root = self.harness.get_filesystem_root("gnbsim")
         (root / "etc/gnbsim/gnb.conf").write_text(read_file("tests/unit/expected_config.yaml"))
         stderr = "whatever stderr content"
         event = Mock()
-        patch_exec.side_effect = ExecError(command=[""], exit_code=1, stderr=stderr, stdout="")
+
+        def gnbsim_handler(_: testing.ExecArgs) -> testing.ExecResult:
+            return testing.ExecResult(stderr=stderr, exit_code=1)
+
+        self.harness.handle_exec("gnbsim", ["/bin/gnbsim"], handler=gnbsim_handler)
         self.harness.set_can_connect(container="gnbsim", val=True)
 
         self.harness.charm._on_start_simulation_action(event=event)
 
         event.fail.assert_called_with(message=f"Failed to execute simulation: {stderr}")
 
-    @patch("ops.model.Container.exec")
     def test_given_simulation_command_fails_with_changeerror_when_start_simulation_action_then_event_fails(  # noqa: E501
-        self, patch_exec
+        self,
     ):
         self.harness.add_storage("config", attach=True)
         root = self.harness.get_filesystem_root("gnbsim")
         (root / "etc/gnbsim/gnb.conf").write_text(read_file("tests/unit/expected_config.yaml"))
         error = "whatever error content"
         event = Mock()
-        patch_exec.side_effect = ChangeError(err=error, change=None)
+
+        def gnbsim_handler(_: testing.ExecArgs) -> testing.ExecResult:
+            raise ChangeError(err=error, change=None)
```

Comment:
change=None fails in static checks because of unexpected arg-type

Reply:
It should be fixed now.